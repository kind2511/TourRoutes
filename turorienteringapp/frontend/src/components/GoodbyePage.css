/* GoodbyePage.css */
.goodbye-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100vh; /* Full viewport height */
    text-align: center;
  }
  
  .goodbye-container h1 {
    font-size: 2rem;
    color: #333;
  }
  