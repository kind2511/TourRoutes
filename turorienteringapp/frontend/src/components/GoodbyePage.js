import React, { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import './GoodbyePage.css'; 

const GoodbyePage = () => {
  const navigate = useNavigate();

  useEffect(() => {
    const timer = setTimeout(() => {
      navigate('/login');
    }, 5000); // Wait for 5  seconds the direct to login page

    return () => clearTimeout(timer); // Clear the timer if the component unmounts
  }, [navigate]);

  return (
    <div className="goodbye-container">
      <h1>Sorry to see you go :(</h1>
      <p>Hope we see you again :)</p>
    </div>
  );
};

export default GoodbyePage;
