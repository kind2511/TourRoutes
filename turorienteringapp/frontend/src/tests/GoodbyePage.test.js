import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import GoodbyePage from '../components/GoodbyePage';

test('GoodbyePage component renders without crashing', () => {
  render(
    <BrowserRouter>
      <GoodbyePage />
    </BrowserRouter>
  );
});

//------------------------------------------>

// Snapshot test for the GoodBye  component
test('HomePage  component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <GoodbyePage  />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
