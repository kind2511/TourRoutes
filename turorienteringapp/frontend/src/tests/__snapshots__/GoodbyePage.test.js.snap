// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage  component snapshot 1`] = `
<DocumentFragment>
  <div
    class="goodbye-container"
  >
    <h1>
      Sorry to see you go :(
    </h1>
    <p>
      Hope we see you again :)
    </p>
  </div>
</DocumentFragment>
`;
